subprojects/docs/src/docs/userguide/migrating_from_maven.adoc

```diff
@@ -0,0 +1,588 @@
+
+Why does this matter? Because many plugins rely on standard Java conventions, so migration is just a matter of replicating the configuration of the Maven plugin in Gradle.
+As an example, here's a simple Maven Checkstyle plugin configuration:
+
+[source,xml]
+----
+...
+<plugin>
+  <groupId>org.apache.maven.plugins</groupId>
+  <artifactId>maven-checkstyle-plugin</artifactId>
+  <version>2.17</version>
+  <executions>
+    <execution>
+      <id>validate</id>
+      <phase>validate</phase>
+      <configuration>
+        <configLocation>checkstyle.xml</configLocation>
+        <encoding>UTF-8</encoding>
+        <consoleOutput>true</consoleOutput>
+        <failsOnError>true</failsOnError>
+        <linkXRef>false</linkXRef>
+      </configuration>
+      <goals>
+        <goal>check</goal>
+      </goals>
+    </execution>
+  </executions>
+</plugin>
+...
+----
+
+Everything outside of the configuration block can safely be ignored when migrating to Gradle.
+In this case, the corresponding Gradle configuration looks like the following:
+
+.Configuring the Gradle Checkstyle Plugin
+====
+include::sample[dir="userguide/mavenMigration/basic/groovy",files="build.gradle[tags=checkstyle]"]
+include::sample[dir="userguide/mavenMigration/basic/kotlin",files="build.gradle.kts[tags=checkstyle]"]
+====
+
+The Checkstyle tasks are automatically added as dependencies of the `check` task, which also includes `test`.
+If you want to ensure that Checkstyle runs before the tests, then just specify an ordering with the mustRunAfter() method:
+
+.Controlling when the `checkstyle` task runs
+====
+include::sample[dir="userguide/mavenMigration/basic/groovy",files="build.gradle[tags=depends-on]"]
+include::sample[dir="userguide/mavenMigration/basic/kotlin",files="build.gradle.kts[tags=depends-on]"]
+====
+
+As you can see, the Gradle configuration is often much shorter than the Maven equivalent.
+You also have a much more flexible execution model since you are no longer constrained by Maven's fixed phases.
+
+While migrating a project from Maven, don't forget about source sets.
+These often provide a more elegant solution for handling integration tests or generated sources than Maven can provide, so you should factor them into your migration plans.
+
+
+=== Ant goals
+
+Many Maven builds rely on the AntRun plugin to customize the build without the overhead of implementing a custom Maven plugin.
+Gradle has no equivalent plugin because Ant is a first-class citizen in Gradle builds, via the `ant` object.
+For example, you can use Ant's Echo task like this:
+
+.Invoking Ant tasks
+====
+include::sample[dir="userguide/mavenMigration/ant/groovy",files="build.gradle"]
+include::sample[dir="userguide/mavenMigration/ant/kotlin",files="build.gradle.kts"]
+====
+
+Even Ant properties and filesets are supported natively.
+To learn more, see <<ant#ant,Using Ant from Gradle>>.
+
+
+[[migmvn:unnecessary_plugins]]
+== Understanding which plugins you don't need
```

Comment:
There's one common thing I haven't seen covered in this guide: how to create and publish source and javadoc, which Maven does by default.

Reply:
I'll see if I have time to fit this in. I think there may already be some content in the user manual that covers this.